### PR TITLE
17283 value for version custom field not displayed in split view

### PR DESCRIPTION
**Pull request by hschink**
[`* `#17283` Value for custom fields of type version not displayed in split view`](https://community.openproject.org/work_packages/17238)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://hotfix-17283-value-for-version-custom-field-not-displayed-in-sp.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js [hunk 1]
```diff
@@ -223,11 +225,35 @@ angular.module('openproject.workPackages.controllers')
...
+      case VERSION_TYPE:
+        var projectId = $scope.workPackage.props.projectId;
+        var versions = VersionService.getVersions(projectId);
+
+        return $q.when(versions).then(function(value) {
+          var versionTitle = 'Couldn\'t resolve version name';
+          var versionHref = PathHelper.staticBase + PathHelper.versionPath(property.value);
+          var version = null;
+
+          // TODO Replace by Array.prototype.find as soon as PhantomJS supports it
```
Comment (myabc) [inline app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js]: @hschink why not use lodash's [`find`](https://lodash.com/docs#find) until there's a supported native implementation?

Comment (hschink) [inline app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js]: @myabc Thanks for the hint! I wasn't aware of this fine library! :astonished:

File: app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js [hunk 2]
```diff
@@ -223,11 +225,30 @@ angular.module('openproject.workPackages.controllers')
...
 
-  function getCustomPropertyValue(customProperty) {
-    if (!!customProperty.value && customProperty.format === USER_TYPE) {
-      return UserService.getUser(customProperty.value);
-    } else {
-      return CustomFieldHelper.formatCustomFieldValue(customProperty.value, customProperty.format);
+  function getCustomPropertyValue(property) {
+    switch(property.format) {
+      case VERSION_TYPE:
+        var projectId = $scope.workPackage.props.projectId;
```
Comment (ulferts) [inline app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js]: @hschink as this case statement has a high probability of becoming very long, please move this out into a separate method.

File: karma/tests/work_packages/controllers/details-tab-overview-controller-test.js [hunk 3]
```diff
@@ -408,6 +412,39 @@ describe('DetailsTabOverviewController', function() {
...
+        });
+
+        it('sets the custom version link href correctly', function() {
+          expect(customVersion.value.href).to.equal('/versions/1');
+        });
+
+        it('is viewable', function() {
+          expect(customVersion.value.viewable).to.be.truthy;
+        });
+      });
```
Comment (ulferts) [inline karma/tests/work_packages/controllers/details-tab-overview-controller-test.js:504]: @hschink there are no specs here for the not so happy path.

**Comment by hschink** [comment]
@ulferts I amended the changes.

File: app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js [hunk 4]
```diff
@@ -223,14 +225,41 @@ angular.module('openproject.workPackages.controllers')
...
+      case USER_TYPE:
+        return UserService.getUser(property.value);
+      default:
+        return CustomFieldHelper.formatCustomFieldValue(property.value, property.format);
     }
   }
 
+  function setCustomPropertyVersionValue(property) {
+    var versionHref = PathHelper.staticBase + PathHelper.versionPath(property.value);
+    var versionTitle = 'Couldn\'t resolve version name';
```
Comment (ulferts) [inline app/assets/javascripts/angular/work_packages/controllers/details-tab-overview-controller.js]: I18n missing

**Comment by ulferts** [comment]
good except for that tiny detail.

**Pull request merged**